### ISSUE TITLE
Maybe not apply to every repo on GitHub?

### ISSUE THREAD

**Issue by mgwalker** [njbmartin/prhero#6]
It seems a little aggressive.  :)

**Comment by njbmartin** [comment]
It's far from aggressive. It's supposed to be playful 😃

**Comment by njbmartin** [comment]
Are you referring to the ❤️ or the ⭐️ ?

**Comment by libbyschuknight** [comment]
Yeah, I don't feel it is okay that a random ❤️  comes up for a PR. Maybe give people the option to opt-in or opt-out?

**Comment by harryi3t** [comment]
I am more interested in what is your bot does or will do in future. The readme is empty.

**Comment by nbuechler** [comment]
It definitely startled me, haha. :heart:

**Comment by libbyschuknight** [comment]
True, re opting out @harryi3t 
Maybe if there was info in the reademe about what the aim of the project is?? That would be a good start. Although it does feel a bit intrusive.

**Comment by JasonLG1979** [comment]
It should be opt-in only, otherwise it's spam.

**Comment by JasonLG1979** [comment]
https://help.github.com/articles/github-terms-of-service/#a-account-terms

Specifically:
"You must be a human. Accounts registered by "bots" or other automated methods are not permitted."

**Comment by JasonLG1979** [comment]
And...
https://help.github.com/articles/github-terms-of-service/#b-api-terms

Specifically:
"Abuse or excessively frequent requests to GitHub via the API may result in the temporary or permanent suspension of your account's access to the API. GitHub, in its sole discretion, will determine abuse or excessive usage of the API. GitHub will make a reasonable attempt via email to warn the account owner prior to suspension."

**Comment by hairmare** [comment]
Without opt-in, both ❤️ and ⭐️ are aggressive and intrusive. They add noise to a platform with an otherwise great SNR. Please deactivate this for the time being or people will probably start reporting the @prhero user as abusive (if they haven't already).

**Comment by sigu** [comment]
Please make this an opt-in bot

**Comment by ehx-v1** [comment]
How about having an actual rating mechanics behind it? It feels too random that all PRs have a heart emoji, but if they have a meaning things quickly get different. For example, a heart for good PRs, a star for really outstanding PRs. Even if "good PR" is as simple as having a description.

**Comment by judwhite** [comment]
Sorry dude. The <3 made me smile (before I knew it was a bot) but @hairmare is right, it's noise.

**Comment by ehx-v1** [comment]
How about my idea? It's noise mainly because there's no meaning behind it.
BTW @njbmartin I'm also planning to help on giving it a meaning once I have time.

**Comment by njbmartin** [comment]
@prhero is now turned off for the time being and may return with your advise of being **opt-in** only.

Here's some information about @prhero.

# What is @prhero?

@prhero is a GitHub experiment designed with good intentions to spread a little ❤️ . During the last 5 hours, it has given ❤️  to over 1.8k public repos.

# Why is it reacting to every PR with a ❤️?

I've seen PRs that people have poured their time and effort into public repos and open source projects, only for it to be ignored or rejected. The intention behind @prhero was supposed to be harmless fun to make people smile, the idea being that somebody out there cares about your hard work.

# But why a ❤️?

@prhero was originally planned to 👍 instead but a ❤️ seemed more harmless and appropriate for the bot. As mentioned above, the sole intention was to spread a little ❤️ and make people smile. PRs don't tend to get anything other than a 👍 

# But it's reacting to all my private PRs!

Actually, @prhero only has access to publicly available repos, which means @prhero can't react to private repos. If your repository is meant to be private, you should think about doing so.

# Isn't this abusing, aggressive and noise? 

A reaction such as this is supposed to be harmless. A few of you have suggested that it's actions are aggressive or abusive, and adds a lot of noise to the platform. I feel that this definitely misinterprets the intentions of @prhero.

Abuse and aggression would suggest @prhero was designed with with bad intentions, but it's not stealing your data, or doing anything negative... unlike the bots on GitHub you don't know about. Reactions were created by GitHub to remove the noise of 👍 comments.

You also don't get notifications when someone reacts to your PR or stars your repo, so there's nothing abusive about it. Were you to get notifications, then absolutely this would be abuse and aggressive.

If it was to comment on every PR instead, that would be abusive.

# But it adds no meaning to the PR discussion

What is love? (Baby don't hurt me, don't hurt me no more). See everything I've said above.

# How does it work?

It's a Node.js server that's polling the GitHub Events API and filtering **opened** pull requests. It then makes a subsequent call to each PR to add a ❤️. It adheres to the rate limit, so it doesn't abuse the API.

# Isn't this against the terms of GitHub?

While @JasonLG1979 has pointed out above that an account must be human, the second line defines that as _Accounts registered by "bots" or other automated methods are not permitted_. As the @prhero account was created by me, it doesn't break that rule. However, the second point regarding "Abuse or excessive use of the API" _may_ apply, but it still adheres to the API rate-limit.

Also, a bot account is how most GitHub integrations work. (eg. @gitter-badger)

# How long was this supposed to last?

The intention was never to keep it constantly online for the foreseeable future. It was designed as a short-lived experiment that would last no longer than a week.

However some people have suggested both improvements as well as ways to make it more meaningful. I hear you, so it may come back online as an opt-in integration powered by web hooks.

**Comment by Vdauphin** [comment]
>  The intention behind @prhero was supposed to be harmless fun to make people smile, the idea being that somebody out there cares about your hard work.

Who want to be like/love by a bot?

**Comment by njbmartin** [comment]
@prhero - The hero we deserve but not the one we need.

**Comment by eed3si9n** [comment]
Initially it might look cute and fun (like the smiling employee of the first McDonald's in Moscow), but the symbol of ❤️ used to mean love, and now it means "really like it", and by giving it out without thoughts eventually even the current meaning will be lost.
If you listen to http://www.npr.org/programs/invisibilia/481887848/the-new-norm it talks about creating dissonance for the McDonald's employee because even when the customers are rude, they *have to* continue to smiling, further encouraging abuse. So my suggestion would be cut the experiment before the inflation, or actually go around the repos and hit ❤️ when you (or a really clever bot) mean it.

**Comment by mgwalker** [comment]
>A few of you have suggested that it's actions are aggressive or abusive, and adds a lot of noise to the platform. I feel that this definitely misinterprets the intentions of @prhero.

The intention matters far less than the actual effect.  I think it's a clever idea, but it's invasive and noisy right now.  It adds content to my repos that doesn't have any meaning - a bot can't actually "like" a thing.  Being opt-in would help, though it dramatically reduces the reach of the bot.  I'm glad you're taking all these comments to heart and I hope you'll have success finding a happy opt-in balance for the bot.  😃

**Comment by njbmartin** [comment]
> I'm glad you're taking all these comments to heart and I hope you'll have success finding a happy opt-in balance for the bot.

Absolutely! It got people talking. Some mixed opinions about it, which is great as it's also sparked some ideas about a non-intrusive version and giving it a proper meaning such as the suggestions from @ehx-v1.

**Comment by ianwremmel** [comment]
Is prhero still turned off? I'd like to opt-out as well. I just spent about 10 minutes investigating why my project was getting spammed.

**Comment by simplesteph** [comment]
https://github.com/njbmartin/prhero/pull/14

**Comment by kalefranz** [comment]
Please stop spamming repos that have not opted in.

**Comment by justmoon** [comment]
You can manually block "prhero" and other spammers from your repos and orgs:

* Block a user from [your repos](https://github.com/settings/blocked_users)
* Block a user from [your orgs](https://help.github.com/articles/blocking-a-user-from-your-organization/)

Unfortunately, this does not remove existing spam from your PRs.

The only other option is **[reporting the issue to Github](https://github.com/contact/report-abuse)**.

**Comment by inket** [comment]
Please report @prhero

**Comment by kalefranz** [comment]
@prhero is not turned off. I'm still being spammed.

**Comment by mmoll** [comment]
please stop this.

**Comment by njbmartin** [comment]
See #16. As well as #17 and #11

**Comment by BBasile** [comment]
Your bot is stupid. It doesn't get anything. If a PR doesnt pass the continuous integration tests ( :x: ) it still be marked as :heart:, e.g like if an user who doesn't get anything is more able to estimate the quality of the PR.

**Comment by mgwalker** [comment]
Great, looks like someone cloned it and is now running another version of it called @broid-love...  Wonderful.  And whoever did that wasn't kind enough to link to their personal profile so people can complain.  😛

**Comment by ickc** [comment]
> You can manually block "prhero" and other spammers from your repos and orgs:
> 
> Block a user from your repos
> Block a user from your orgs
> Unfortunately, this does not remove existing spam from your PRs.

Would deleting the account @prhero removes the "spam" already done?

If so, just delete the account and reopen it.
